### PR TITLE
Show default patterns and products if the images request fails

### LINKED ISSUE #42994: CYS > Ensure the user is led to the Assembler with the default content whenever the Pexels API is down on initial store setup
## **Issue Description:**
While it is rare for the Pexels API to be down, it is not impossible. With that said, let's make sure that whenever the store owner receives the following response from Pexels **when creating their store for the very first time**, we allow proceeding to the Assembler and display the patterns and dummy products with their default content instead:

```
WP_Error Object
(
    [errors] => Array
        (
            [pexels_api_error] => Array
                (
                    [0] => Request to the Pexels API failed.
                )

        )

    [error_data] => Array
        (
            [pexels_api_error] => Array
                (
                    [code] => 500
                    [data] => Array
                        (
                            [code] => rest_request_error
                            [message] => An unknown error has occurred. Please try again later.
                            [data] => Array
                                (
                                    [status] => 500
                                )

                        )

                )

        )

    [additional_data:protected] => Array
        (
        )

)
```

### PR DESCRIPTION
**Pull request by albarin**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR resets patterns and products to default when the request to the images endpoint (Pexels API) fails, only the first time (when the `is_ai_generated` is false).

Closes https://github.com/woocommerce/woocommerce/issues/42994

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:
**Initial Setup**
1. Create a new WooCommerce installation with this version.
2. Ensure Jetpack is installed and your site is connected with JP.
3. Ensure the WooCommerce Beta Tester plugin is installed and activated (available on this monorepo).
4. Ensure the Woo AI plugin is installed and activated (available on this monorepo).
5. Head over to `/wp-admin/tools.php?page=woocommerce-admin-test-helper` and enable `customize-store` feature flag:

**Testing steps**
1. Modify the `request` function on the `plugins/woocommerce/src/Blocks/Images/Pexels.php` class to return an error (to simulate an error from the Pexels API):
```
private function request( string $search_term, int $per_page = 100 ) {
                return new \WP_Error( 'pexels_api_error', __( 'Request to the Pexels API failed.', 'woocommerce' ), $error_msg );
```
2. Head over to WooCommerce -> Home and click on "Start Customizing".
3. Click the Design with AI button.
4. Fill out a business description and proceed.
5. Make sure the store is generated and the patterns and products have the default content.
6. Remove the line of code added in step 1.
7. Repeat the process and make sure this time the result is AI-generated.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Ensure the user is led to the Assembler with the default content whenever the Pexels API is down on the initial store setup.

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/customize-store/design-with-ai/services.ts [hunk 1]
```diff
@@ -220,6 +248,18 @@ export const updateStorePatterns = async (
...
 		} );
 
+		const { is_ai_generated } = await apiFetch< {
+			is_ai_generated: boolean;
+		} >( {
+			path: '/wc/private/ai/store-info',
+			method: 'GET',
+		} );
+
+		if ( ! images.length && ! is_ai_generated ) {
```
Comment (gigitux) [inline plugins/woocommerce-admin/client/customize-store/design-with-ai/services.ts]: We do the check `is_ai_generated` inside the `resetPatternsAndProducts` function. I think that this is enough:
```suggestion
		if ( ! images.length) {
```

**Review by gigitux — APPROVED** [review]
LGTM! I suggested a change, but feel free to skip if you think that it is not necessary.

File: plugins/woocommerce-admin/client/customize-store/design-with-ai/services.ts (hunk 1 above)
Comment (albarin) [reply plugins/woocommerce-admin/client/customize-store/design-with-ai/services.ts]: You are right! Updating, thanks!! 🙌

File: plugins/woocommerce-admin/client/customize-store/design-with-ai/services.ts [hunk 2]
```diff
@@ -220,6 +248,11 @@ export const updateStorePatterns = async (
 			},
 		} );
 
+		if ( ! images.length ) {
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/customize-store/design-with-ai/services.ts:251]: Hey @albarin 👋  I just tested the trunk branch on a JN site and couldn't generate content/images with AI.

It seems that `/wc/private/ai/images` endpoint returned a different format than expected. Could you take a look?

Screenshot 2024-01-04 at 21 02 17

Comment (albarin) [reply plugins/woocommerce-admin/client/customize-store/design-with-ai/services.ts:251]: That's weird, I'll have a look, it was testing fine in WooExpress 🤔

Comment (albarin) [reply plugins/woocommerce-admin/client/customize-store/design-with-ai/services.ts:251]: I have opened a PR to address this, thanks for spotting this https://github.com/woocommerce/woocommerce/pull/43285

**Pull request merged**